src/execution/as_abi.rs

```diff
@@ -269,24 +302,28 @@ pub(crate) fn assembly_script_append_data(env: &ASEnv, key: i32, value: i32) ->
 }
 
 /// gets a key-indexed data entry in the datastore, failing if non-existent
-pub(crate) fn assembly_script_get_data(env: &ASEnv, key: i32) -> ABIResult<i32> {
-    sub_remaining_gas(env, settings::metering_get_data_const())?;
+pub(crate) fn assembly_script_get_data(mut ctx: FunctionEnvMut<ASEnv>, key: i32) -> ABIResult<i32> {
+
+    let env = ctx.data().clone();
```

Comment:
In a lot of functions each time you pass an `env` you also pass the `ctx` that contains this env. Is that possible to not make this `clone` and just use the `env` of `ctx` on each function ?